include/runtime/collections/rangemap.h

```diff
@@ -0,0 +1,380 @@
+#ifndef RANGEMAP_HEADER_H
+#define RANGEMAP_HEADER_H
+
+#include "RBTree.h"
+
+#include <cassert>
+#include <iostream>
+#include <optional>
+#include <stdexcept>
+
+template <class T, class V>
+class RangeMap {
+public:
+  using Range = std::pair<T, T>;
```

Comment:
There's enough functionality hooked into `Range` below that I would consider extracting it out into its own class; this makes your test + user code for the rangemap a bit nicer to read as well.

Reply:
Range is not extracted into its own class.